From penberg@cs.helsinki.fi  Mon Jan 16 14:05:27 2006
From: Pekka Enberg <penberg@cs.helsinki.fi>
Subject: USB: remove LINUX_VERSION_CODE macro usage
To: Greg KH <gregkh@suse.de>
Date: Thu, 12 Jan 2006 20:35:25 +0200
Message-Id: <1137090925.9933.0.camel@localhost>

From: Pekka Enberg <penberg@cs.helsinki.fi>

This patch removes unnecessary LINUX_VERSION_CODE macro usage from
drivers/usb/.

Signed-off-by: Pekka Enberg <penberg@cs.helsinki.fi>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>
---
 drivers/usb/media/pwc/pwc-ctrl.c    |    1 -
 drivers/usb/media/pwc/pwc-if.c      |    6 +-----
 drivers/usb/misc/sisusbvga/sisusb.h |    8 --------
 3 files changed, 1 insertion(+), 14 deletions(-)

--- gregkh-2.6.orig/drivers/usb/media/pwc/pwc-ctrl.c
+++ gregkh-2.6/drivers/usb/media/pwc/pwc-ctrl.c
@@ -41,7 +41,6 @@
 #include <asm/uaccess.h> 
 #endif
 #include <asm/errno.h>
-#include <linux/version.h>
  
 #include "pwc.h"
 #include "pwc-ioctl.h"
--- gregkh-2.6.orig/drivers/usb/media/pwc/pwc-if.c
+++ gregkh-2.6/drivers/usb/media/pwc/pwc-if.c
@@ -62,7 +62,6 @@
 #include <linux/poll.h>
 #include <linux/slab.h>
 #include <linux/vmalloc.h>
-#include <linux/version.h>
 #include <asm/io.h>
 
 #include "pwc.h"
@@ -827,13 +826,10 @@ static int pwc_isoc_init(struct pwc_devi
 	/* Get the current alternate interface, adjust packet size */
 	if (!udev->actconfig)
 		return -EFAULT;
-#if LINUX_VERSION_CODE < KERNEL_VERSION(2,6,5)
-	idesc = &udev->actconfig->interface[0]->altsetting[pdev->valternate];
-#else
+
 	intf = usb_ifnum_to_if(udev, 0);
 	if (intf)
 		idesc = usb_altnum_to_altsetting(intf, pdev->valternate);
-#endif
 		
 	if (!idesc)
 		return -EFAULT;
--- gregkh-2.6.orig/drivers/usb/misc/sisusbvga/sisusb.h
+++ gregkh-2.6/drivers/usb/misc/sisusbvga/sisusb.h
@@ -37,24 +37,16 @@
 #ifndef _SISUSB_H_
 #define _SISUSB_H_
 
-#include <linux/version.h>
 #ifdef CONFIG_COMPAT
-#if LINUX_VERSION_CODE <= KERNEL_VERSION(2,6,10)
-#include <linux/ioctl32.h>
-#define SISUSB_OLD_CONFIG_COMPAT
-#else
 #define SISUSB_NEW_CONFIG_COMPAT
 #endif
-#endif
 
 /* For older kernels, support for text consoles is by default
  * off. To ensable text console support, change the following:
  */
 #if 0
-#if LINUX_VERSION_CODE <= KERNEL_VERSION(2,6,13)
 #define CONFIG_USB_SISUSBVGA_CON
 #endif
-#endif
 
 /* Version Information */
 
